src/adapter/tracer/mod.rs

```diff
@@ -0,0 +1,4 @@
+pub mod ptrace;
```

Comment:
I'm not necessarily convinced of the value of the extra `pub` layer of indirection here.

Perhaps `mod ptrace;` plus re-exporting what's intended to be `pub` would suffice?